android/app/src/main/java/org/mozilla/magnet/database/HistoryDatabaseSQL.java

```diff
@@ -0,0 +1,158 @@
+     * giving us the opportunity to migrate tables to
+     * a new schema.
+     *
+     * Right now we're simply dropping the old table
+     * and creating a new replacement.
+     *
+     * @param db
+     * @param oldVersion
+     * @param newVersion
+     */
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.execSQL("DROP TABLE IF EXISTS " + HistoryEntry.TABLE_NAME);
+        onCreate(db);
+    }
+
+    /**
+     * Insert a new record.
+     *
+     * @param url
+     */
+    public void insert(String url) {
+        SQLiteDatabase db = this.getWritableDatabase();
+        long now = System.currentTimeMillis() / 1000;
+
+        ContentValues values = new ContentValues();
+        values.put(COLUMN_NAME_URL, url);
+        values.put(HistoryEntry.COLUMN_NAME_TIME_FIRST_SEEN, now);
+        values.put(COLUMN_NAME_TIME_LAST_SEEN, now);
+
+        db.insert(HistoryEntry.TABLE_NAME, null, values);
+        db.close();
+    }
+
+    /**
+     * Update the last-seen column for a previously
+     * recorded entry.
+     *
+     * @param id
+     */
+    public void updateLastSeen(int id) {
+        Log.d(TAG, "update last seen: " + id);
+        long now = System.currentTimeMillis() / 1000;
+        ContentValues values = new ContentValues();
+        values.put(COLUMN_NAME_TIME_LAST_SEEN, now);
+        update(id, values);
+    }
+
+    /**
+     * Update a row with the given values.
+     *
+     * @param id
+     * @param values
+     */
+    private void update(int id, ContentValues values) {
+        SQLiteDatabase db = this.getWritableDatabase();
+
+        db.update(
+                HistoryEntry.TABLE_NAME,
+                values,
+                HistoryEntry._ID + " = ?",
+                new String[]{String.valueOf(id)});
+
+        db.close();
+    }
+
+    /**
+     * Attempt to find a single row with the given url
+     * that was last seen since the given date.
+     *
+     * @param url
+     * @param date
+     * @return HistoryRecord
+     */
+    public HistoryRecord getSince(String url, Date date) {
+        String query = "SELECT * FROM " + HistoryEntry.TABLE_NAME
+                + " WHERE " + COLUMN_NAME_TIME_LAST_SEEN + " > " + date.getTime() / 1000
+                + " AND " + COLUMN_NAME_URL + " = '" + url + "'"
+                + " LIMIT 1";
+
+        SQLiteDatabase db = this.getWritableDatabase();
+        Cursor cursor = db.rawQuery(query, null);
+
+        // return null if nothing was found
+        if (!cursor.moveToFirst()) {
+            cursor.close();
+            return null;
+        }
+
+        HistoryRecord result = toHistoryRecord(cursor);
+        cursor.close();
+        return result;
+    }
+
+    /**
+     * Return a HistoryRecord from the given cursor.
+     *
+     * @param cursor
+     * @return HistoryRecord
+     */
+    private HistoryRecord toHistoryRecord(Cursor cursor) {
+        int id = cursor.getInt(cursor.getColumnIndex(_ID));
```

Comment:
If it's needed in the future yes. Instead of changing this, I would create a second function that calls `toHistoryRecord()` for each item in the list.